amcl/src/amcl_node.cpp

```diff
@@ -959,6 +1009,9 @@ AmclNode::handleMapMessage(const nav_msgs::OccupancyGrid& msg)
   // try to apply the initial pose now that the map has arrived.
   applyInitialPose();
```

Comment:
In apply_initial_pose function particle filter is initialized using  initial_poses_hyp and in updatePoseFromServer function inital_pose variable is updated.   
I feel we need to update initial_poses_hyp as well in order to apply the pose correctly.

pf_init(pf_, initial_poses_hyp_.size(), &initial_means[0], &initial_covs[0]);

Reply:
we don't need to initialize initial_poses_hyp_ as it  applyInitialPose clears it up and will just exit